db/migrate/20200331111947_add_reporting_org_association_to_activity.rb

```diff
@@ -0,0 +1,6 @@
+class AddReportingOrgAssociationToActivity < ActiveRecord::Migration[6.0]
+  def change
+    remove_column :activities, :reporting_organisation_reference, :string
```

Comment:
I'm nervous that the column us removed as a separate task to the one that preserves the existing data. I'm not sure of the strategy that would allow us to use them both to switch over to the new method?

I think there are at least 2 options to help us. The easier one is probably to use migrations for this to change both schema and data in one:

1. add new column
1. start a database transaction
1. migrate data into new column 
1. end database transaction
1. delete old column

Our down operation can be coded to take the iati_reference from the record and save the string too. 

An advantage of this is that we will not have to remember to run the rask task on each environment in the correct sequencing of migrations. When migrations are run the data is switched over as part of it automatically.

[Here's an example I dug up from MyTVH](https://github.com/dxw/mytvh-web/blob/develop/db/migrate/20140619111622_change_force_changed_passphrase_for_user.rb) which illustrates what I'm thinking of.

Reply:
Good call 👍